resample/empirical.py

```diff
@@ -0,0 +1,80 @@
+"""
+Various empirical functions.
+
+Empirical means they are based on the original data sample instead of a parameteric
+density function.
+"""
+from typing import Callable, Sequence
+import numpy as np
+from scipy.interpolate import interp1d as _interp1d
+from resample.jackknife import jackknife as _jackknife
+
+
+def cdf_gen(sample: np.ndarray) -> Callable:
```

Comment:
I think this could just be `cdf`?

Reply:
It returns a function, you said before you wanted those to be called "X_fn". If that is called just `cdf`, then also `quantile_fn` should just be `quantile`. But I think you were right that higher-order functions which return other functions should have a different name than ordinary functions.

Reply:
It stands for cumulative distribution function so "function" is already in the name

Reply:
Yes, but `cdf_gen` creates a cdf, it does not return the value computed by the cdf. It is a cdf generator / higher-order function / meta-function, whatever you prefer to call it. Likewise for `quantile`.

The usual naming conventions in programming is that we do not call something a "function" or a "mapping" explicitly in the function name. Therefore, something that computes the cumulative density would normally be called `cumulative_density` or `cd`, not `cumulative_density_function` or `cdf`. However, since `cdf` is an abbreviation that everyone recognizes, libraries usually chose `cdf`, making the "function" part of the name, although it goes against normal conventions.

This is the situation we are facing here. Consistency is important. In scipy, norm.cdf(...) returns the cumulative density for the argument, not the cdf. norm.cdf is a normal function. Our `cdf_gen` is a metafunction. We want to make that clear in the name. It is therefore not superfluous to add `_gen` or `_fn` to indicate that this is a function generator.

Having said all that, I think the suffix `cdf_gen` makes that distinction more clear than `cdf_fn`.

Reply:
In C++, could be considered a factory `cdf_gen`, and then an appropriate name would be `make_cdf`, but I don't think this is a good match for Python.

Reply:
I was actually thinking in terms of the name of the thing outside the world of programming and not the return type when suggesting we use a name like quantile_function (communicating that it's a higher-order function in my mind is the purpose of documentation / type annotations and generally wouldn't be the responsibility of the function name, similar to how we wouldn't name something quantile_float if it was returning a single quantile).

My personal opinion and maybe we just merge for now and reconsider later.

Reply:
Ok, let's discuss this later.